src/MultiDimFit.cc

```diff
@@ -665,42 +680,148 @@ void MultiDimFit::doGrid(RooWorkspace *w, RooAbsReal &nll)
+                    }
+                }
+            }
+
+            // Loop over starting points to try for the prof WCs
+            for (unsigned int start_pt_idx=0; start_pt_idx<wc_vals_vec_of_vec.size(); start_pt_idx++){
+                *params = snap;
+                poiVals_[0] = x;
+                poiVars_[0]->setVal(x);
+
+                // Loop over prof POIs and set their values
+                if (verbose > 1) std::cout << "\n\tStart pt idx: " << start_pt_idx << std::endl;
+                for (unsigned int var_idx=0; var_idx<specifiedVars_.size(); var_idx++){
+                    if (verbose > 1) std::cout << "\t\tThe var name: " << specifiedVars_[var_idx]->GetName() << std::endl;
+                    if (strcmp(specifiedVars_[var_idx]->GetName(),"r")==0) {
+                        // Don't bother setting r to anything
+                        if (verbose > 1) std::cout << "\t\t\tSkipping var " << specifiedVars_[var_idx]->GetName() << std::endl;
+                        continue;
+                    }
+                    if (verbose > 1) std::cout << "\t\t\tRange before: " << specifiedVars_[var_idx]->getMin() << " " << specifiedVars_[var_idx]->getMax() << std::endl;
+                    if (verbose > 1) std::cout << "\t\t\t" << specifiedVars_[var_idx]->GetName() << " before setting: " << specifiedVars_[var_idx]->getVal() << " += " << specifiedVars_[var_idx]->getError() << std::endl;
+                    if (pointsRandProf_ > 0) {
+                        specifiedVars_[var_idx]->removeRange(); // Do not impose a range if we're trying multiple random start points
+                    }
+                    specifiedVars_[var_idx]->setVal(wc_vals_vec_of_vec.at(start_pt_idx).at(var_idx));
+                    if (verbose > 1) std::cout << "\t\t\tRange after: " << specifiedVars_[var_idx]->getMin() << " " << specifiedVars_[var_idx]->getMax() << std::endl;
+                    if (verbose > 1) std::cout << "\t\t\t" << specifiedVars_[var_idx]->GetName() << " after  setting: " << specifiedVars_[var_idx]->getVal() << " += " << specifiedVars_[var_idx]->getError() << std::endl;
+                }
+
+                // now we minimize
+                nll.clearEvalErrorLog();
+                deltaNLL_ = nll.getVal() - nll0;
+                if (nll.numEvalErrors() > 0) {
+                    deltaNLL_ = 9990;
+                    for(unsigned int j=0; j<specifiedNuis_.size(); j++){
+                        specifiedVals_[j]=specifiedVars_[j]->getVal();
+                    }
+                    for(unsigned int j=0; j<specifiedFuncNames_.size(); j++){
+                        specifiedFuncVals_[j]=specifiedFunc_[j]->getVal();
+                    }
+                    Combine::commitPoint(true, /*quantile=*/0);
+                    continue;
+                }
+                ok = fastScan_ || (hasMaxDeltaNLLForProf_ && (nll.getVal() - nll0) > maxDeltaNLLForProf_) || utils::countFloating(*params)==0 ?
```

Comment:
Just from looking at this I think this instance of `ok` is not used at all as L803 is always invoked afterwards, thus no need to set it here?